python/pyarrow/array.pxi

```diff
@@ -3519,17 +3519,42 @@ class FixedShapeTensorArray(ExtensionArray):
 
     def to_numpy_ndarray(self):
```

Comment:
Why isn't it called `to_numpy` and `from_numpy` btw?

Reply:
`to_numpy`
```python
import pyarrow as pa
arrow_type = pa.int8()
tensor_type = pa.fixed_shape_tensor(arrow_type, [2, 3])
storage = pa.array([[1, 2, 3, 4, 5, 6], [7, 8, 9, 10, 11, 12]], pa.list_(arrow_type, 6))
arr = pa.ExtensionArray.from_storage(tensor_type, storage)
arr.to_numpy(zero_copy_only=False)
```
already returns storage arrays:
```
array([array([1, 2, 3, 4, 5, 6], dtype=int8),
       array([ 7,  8,  9, 10, 11, 12], dtype=int8)], dtype=object)
```
We agreed (with @jorisvandenbossche  and @AlenkaF ), in a thread I can't find now, to use `to_numpy_ndarray` for getting tensors from `FixedShapeTensorArray` into numpy. I'd use the same convention for `from_numpy_ndarray` to go from `ndarray` into `FixedShapeTensorArray`

Reply:
> ```
> array([array([1, 2, 3, 4, 5, 6], dtype=int8),
>        array([ 7,  8,  9, 10, 11, 12], dtype=int8)], dtype=object)
> ```

In which circumstances is this a desirable behavior? It even makes a copy, right?

Reply:
I wouldn't know, this is defined in [Array class](https://github.com/apache/arrow/blob/main/python/pyarrow/array.pxi#L1537) and I think we just didn't want to override it.

Reply:
I really think we should change `to_numpy` in this case. The current behavior does not seem useful.

Reply:
Ok, we can make this a separate issue since the function already exists.

Reply:
Opened https://github.com/apache/arrow/issues/39991.